packages/application-generic/src/services/filters/filter-conditions.service.ts

```diff
@@ -0,0 +1,636 @@
+import axios from 'axios';
+import {
+  parseISO,
+  differenceInMinutes,
+  differenceInHours,
+  differenceInDays,
+} from 'date-fns';
+import { Injectable } from '@nestjs/common';
+import {
+  EnvironmentRepository,
+  ExecutionDetailsRepository,
+  JobEntity,
+  JobRepository,
+  MessageRepository,
+  StepFilter,
+  SubscriberEntity,
+  SubscriberRepository,
+} from '@novu/dal';
+import {
+  ChannelTypeEnum,
+  FieldLogicalOperatorEnum,
+  FieldOperatorEnum,
+  FilterParts,
+  FilterPartTypeEnum,
+  FILTER_TO_LABEL,
+  ICondition,
+  IOnlineInLastFilterPart,
+  IPreviousStepFilterPart,
+  IRealtimeOnlineFilterPart,
+  IWebhookFilterPart,
+  PreviousStepTypeEnum,
+  TimeOperatorEnum,
+} from '@novu/shared';
+import { EmailEventStatusEnum } from '@novu/stateless';
+
+import {
+  FilterProcessingDetails,
+  FilterService,
+  IFilterData,
+  IFilterVariables,
+} from './filters.service';
+
+import { buildSubscriberKey, CachedEntity } from '../cache';
+import { Instrument } from '../../instrumentation';
+import { createHash } from '../../encryption';
+import {
+  EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER,
+  PlatformException,
+} from '../../utils/exceptions';
+
+export interface IUsedFilters {
+  filters: string[];
+  failedFilters: string[];
+  passedFilters: string[];
+}
+
+export type WebhookPartialPayload = Partial<{
+  subscriber: SubscriberEntity | null;
+  payload: Record<string, unknown>;
+  identifier: string;
+  channel: string;
+  providerId: string;
+}>;
+
+export interface IFilterConditionsResponse {
+  passed: boolean;
+  data: IFilterData;
+  conditions: ICondition[];
+  details: FilterProcessingDetails[];
+  variables: IFilterVariables;
+  usedFilters: IUsedFilters;
+}
+
+export interface IFilterProperties {
+  environmentId?: string;
+  identifier?: string;
+  job?: JobEntity;
+  organizationId?: string;
+  subscriberId?: string;
+  transactionId?: string;
+}
+
+@Injectable()
+export class FilterConditionsService extends FilterService {
+  private environmentRepository: EnvironmentRepository;
+  private executionDetailsRepository: ExecutionDetailsRepository;
+  private jobRepository: JobRepository;
+  private messageRepository: MessageRepository;
+  private subscriberRepository: SubscriberRepository;
+
+  constructor() {
+    super();
+    this.environmentRepository = new EnvironmentRepository();
+    this.executionDetailsRepository = new ExecutionDetailsRepository();
+    this.jobRepository = new JobRepository();
+    this.messageRepository = new MessageRepository();
+    this.subscriberRepository = new SubscriberRepository();
+  }
+
+  public async filter(
```

Comment:
do we have any conventions regarding the main service method? like we have execute for the use case.

Reply:
As far as I know not, mainly because we indistinctly use Service for any class that does any operation all across the different layers and domains, so it is hard to come to a convention. I chose `filter` as it was the closer to the domain operation we are executing.